node/src/effect/incoming.rs

```diff
@@ -0,0 +1,186 @@
+//! Announcements of incoming network messages.
+//!
+//! Any event suffixed -`Incoming` is usually the arrival of a specific network message.
+
+use std::{fmt::Display, sync::Arc};
+
+use datasize::DataSize;
+use serde::Serialize;
+
+use crate::{
+    components::{consensus, gossiper},
+    types::{FinalitySignature, NodeId, Tag},
+};
+
+/// An envelope for an incoming message, attaching a sender address.
+#[derive(DataSize, Debug, Serialize)]
+pub struct MessageIncoming<I, M> {
+    pub(crate) sender: I,
+    pub(crate) message: M,
+}
+
+impl<I, M> Display for MessageIncoming<I, M>
+where
+    I: Display,
+    M: Display,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "incoming from {}: {}", self.sender, self.message)
+    }
+}
+
+/// A new consensus message arrived.
+pub(crate) type ConsensusMessageIncoming<I> = MessageIncoming<I, consensus::ConsensusMessage>;
+
+/// A new message from a gossiper arrived.
+pub type GossiperIncoming<T> = MessageIncoming<NodeId, gossiper::Message<T>>;
+
+/// A new message requesting various objects arrived.
+pub(crate) type NetRequestIncoming = MessageIncoming<NodeId, NetRequest>;
+
+/// A new message requesting a trie arrived.
+pub(crate) type TrieRequestIncoming = MessageIncoming<NodeId, TrieRequest>;
+
+/// A new message requesting various objects arrived.
+pub(crate) type NetResponseIncoming = MessageIncoming<NodeId, NetResponse>;
+
+/// A new message requesting a trie arrived.
+pub(crate) type TrieResponseIncoming = MessageIncoming<NodeId, TrieResponse>;
+
+/// A new finality signature arrived over the network.
+pub(crate) type FinalitySignatureIncoming = MessageIncoming<NodeId, Box<FinalitySignature>>;
+
+/// A request for an object out of storage arrived.
+///
+/// Note: The variants here are grouped under a common enum, since they are usually handled by the
+///       same component. If this changes, split up this type (see `TrieRequestIncoming` for an
+///       example).
+#[derive(DataSize, Debug, Serialize)]
+#[repr(u8)]
+pub(crate) enum NetRequest {
+    /// Request for a deploy.
+    Deploy(Vec<u8>),
+    /// Request for a block.
+    Block(Vec<u8>),
+    /// Request for a gossiped public listening address.
+    // TODO: Move this out of `NetRequest` into its own type, it is never valid.
+    GossipedAddress(Vec<u8>),
+    /// Request for a block by its height in the linear chain.
+    BlockAndMetadataByHeight(Vec<u8>),
```

Comment:
The "metadata" is just the finality signatures, I think? Maybe we should also call this `BlockAndFinalitySignaturesByHeight` then (also in the `Display` impl and `NetResponse`).

Reply:
Probably a good idea, but out of scope for this PR in my opinion. I'd rather not change these here.